include/swift/AST/DiagnosticsSema.def

```diff
@@ -8392,6 +8392,28 @@ ERROR(invalid_function_conversion_with_non_sendable,none,
 NOTE(type_does_not_conform_to_Sendable,none,
      "type %0 does not conform to 'Sendable' protocol", (Type))
 
+GROUPED_WARNING(
+    attr_execution_nonisolated_behavior_will_change_decl, AsyncCallerExecution,
+    none,
+    "feature '%0' will cause nonisolated async %kindbase1 to run in the "
+    "caller's isolation domain; use %2 to preserve behavior",
+    (StringRef, const AbstractFunctionDecl *, DeclAttribute))
+
+GROUPED_WARNING(
+    attr_execution_nonisolated_behavior_will_change_closure,
+    AsyncCallerExecution, none,
+    "feature '%0' will cause nonisolated async closure to run in the caller's "
+    "isolation domain; use %1 to preserve behavior",
```

Comment:
Can we drop "domain"? We kinda casually use it but it's not really a term we actually document, just isolation should be enough?

Reply:
I prefer "on the caller's actor"